glean-core/python/glean/glean.py

```diff
@@ -138,10 +137,7 @@ def initialize(
                 configuration = Configuration()
 
             if data_dir is None:
-                data_dir = Path(tempfile.TemporaryDirectory().name)
-                cls._destroy_data_dir = True
-            else:
-                cls._destroy_data_dir = False
+                raise TypeError("data_dir must be provided")
```

Comment:
If only we had a type system to ensure this :P

Reply:
Yeah -- Python has ended up in a truly strange place with respect to type-checking, but here we are.